delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphIndexingCoordinator.scala

```diff
@@ -100,40 +105,76 @@ private class IndexingStream(
+    new IllegalArgumentException(error.toString)
+
+  private def fetchView(views: BlazegraphViews, config: BlazegraphViewsConfig) = (id: Iri, project: ProjectRef) =>
+    views
+      .fetchIndexingView(id, project)
+      .flatMap { res =>
+        UIO.pure(
+          Some(
+            ViewIndex(
+              res.value.project,
+              res.id,
+              BlazegraphViews.projectionId(res),
+              BlazegraphViews.index(res, config.indexing),
+              res.rev,
+              res.deprecated,
+              res.value
+            )
+          )
+        )
+      }
+      .onErrorHandle {
+        case _: DifferentBlazegraphViewType =>
+          logger.debug(s"Filtering out aggregate views")
+          None
+        case r                              =>
+          logger.error(
+            s"While attempting to start indexing view $id in project $project, the rejection $r was encountered"
+          )
+          None
+      }
+
   /**
     * Create a coordinator for indexing triples into Blazegraph namespaces triggered and customized by the BlazegraphViews.
     */
   def apply(
-      eventLog: BlazegraphIndexingEventLog,
+      views: BlazegraphViews,
+      indexingLog: BlazegraphIndexingEventLog,
       client: BlazegraphClient,
       projection: Projection[Unit],
       cache: ProgressesCache,
       config: BlazegraphViewsConfig
   )(implicit
+      uuidF: UUIDF,
       as: ActorSystem[Nothing],
       scheduler: Scheduler,
       base: BaseUri,
       resolution: RemoteContextResolution
-  ): Task[BlazegraphIndexingCoordinator] = {
-
-    val indexingRetryStrategy =
-      RetryStrategy.retryOnNonFatal(config.indexing.retry, logger, "blazegraph indexing")
-
-    implicit val indexCfg: ExternalIndexingConfig = config.indexing
-
-    IndexingStreamCoordinator[ResourceF[IndexingBlazegraphView]](
-      "BlazegraphViewsCoordinator",
-      (res, progress) => new IndexingStream(client, cache, res, config).build(eventLog, projection, progress),
-      client.deleteNamespace(_).void,
-      projection,
-      config.aggregate.processor,
-      indexingRetryStrategy
-    )
-  }
+  ): Task[BlazegraphIndexingCoordinator] = Task
+    .delay {
+      val indexingRetryStrategy =
+        RetryStrategy.retryOnNonFatal(config.indexing.retry, logger, "blazegraph indexing")
+
+      new IndexingStreamCoordinator[IndexingBlazegraphView](
+        BlazegraphViews.moduleType,
+        fetchView(views, config),
+        (res, progress) => new IndexingStream(client, cache, res, config).build(indexingLog, projection, progress),
+        client.deleteNamespace(_).void,
+        projection,
+        indexingRetryStrategy
+      )
+    }
+    .tapEval { coordinator =>
+      def onEvent(event: BlazegraphViewEvent) = coordinator.run(event.id, event.project, event.rev)
+      IO.unless(MigrationState.isIndexingDisabled)(
+        BlazegraphViewsIndexing("BlazegraphIndexingCoordinatorScan", config.indexing, views, onEvent)
```

Comment:
The reason is that it introduces a circular dependency between the BlazegraphViews and its coordinator that distage was not able to handle (it could not proxy BlazegraphViews, asking if it was final).
So the reason was to break this dependency as we should avoid them when we can.
There would be another option to handle this, but this would require to init the cache in the module and not in BlazegraphViews.

Reply:
If that's the case I would rather have something more explicit in `BlazegraphViewsIndexing`, like:
```scala
def populateCache(name: String, config: ExternalIndexingConfig, views: BlazegraphViews) =
   apply(name, 
   config, 
   views, 
   event  =>
                               views
                                 .fetch(event.id, event.project)
                                 .redeemCauseWith(_ => IO.unit, res => index.put(ViewRef(res.value.project, res.value.id), res))
) 

def triggerCoordinator(name: String, config: ExternalIndexingConfig, coordinator: BlazegraphIndexingCoordinator) =
   apply(name, config, views, event => coordinator.run(event.id, event.project, event.rev))
```

same for elasticsearch